src/App.js

```diff
@@ -0,0 +1,94 @@
+import React, { useState } from "react";
+import "bulma/css/bulma.min.css";
+import "./App.css";
+
+function App() {
+    const INIT_SIZE = 3;
+
+    const initGrid = (size) => {
+        const newGrid = [];
+        for (let x = 0; x < size; x++) {
+            const arr = new Array(size);
+            arr.fill({});
+            newGrid.push(arr);
+        }
+        return newGrid;
+    };
+
+    const initial = {
+        size: INIT_SIZE,
+        grid: initGrid(INIT_SIZE),
+    };
+    const [state, setState] = useState(initial);
+
+    const handleChange = (e) => {
+        let newSize = e.target.value;
+        newSize = parseInt(newSize, 10) > 40 ? 40 : parseInt(newSize, 10);
```

Comment:
Try not to use what they call "magic numbers" ;-)

A magic number is a numeric literal (i.e. it is just a number that has been committed) that does not live in a variable.

It is much more clear to do something like
```
const MAX_SIZE = 40;
...
newSize = parseInt(newSize, 10) > MAX_SIZE ? MAX_SIZE : parseInt(newSize, 10)
```

Reply:
> Try not to use what they call "magic numbers" ;-)
> 
> A magic number is a numeric literal (i.e. it is just a number that has been committed) that does not live in a variable.
> 
> It is much more clear to do something like
> 
> ```
> const MAX_SIZE = 40;
> ...
> newSize = parseInt(newSize, 10) > MAX_SIZE ? MAX_SIZE : parseInt(newSize, 10)
> ```

ok